pkg/telemetry/telemetry.go

```diff
@@ -224,6 +224,14 @@ func (t *TelemetryClient) TrackMetric(metricname string, value float64, properti
 
 func (t *TelemetryClient) TrackTrace(name string, severity contracts.SeverityLevel, properties map[string]string) {
 	trace := appinsights.NewTraceTelemetry(name, severity)
+
+	kernelVersion, err := kernelVersionFunc(context.Background()) 
+	if err == nil {
+		properties[kernelversion] = kernelVersion 
+	} else {
+		t.trackWarning(err, "failed to get kernel version for trace")
+	}
```

Comment:
Usually preference is to test `err != nil`, can you swap this conditional?

Reply:
Hi, @timraymond I've made the necessary changes and pushed the code. Kindly let me know if anything else needs to be changed/modified. Thank you.

Reply:
@Anirudh2112 I think you need to take another look at that conditional and make sure you updated it fully